src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -174,74 +174,60 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
             }
             finally
             {
                 semaphore.Release();
             }
         }
 
-        async Task AcknowledgeMessage(IModel channel, ulong deliveryTag, string messageId)
+        class MessageState
         {
-            try
-            {
-                var task = new Task(() =>
-                {
-                    if (channel.IsOpen)
-                    {
-                        channel.BasicAck(deliveryTag, false);
-                    }
-                    else
-                    {
-                        Logger.Warn($"Attempt to acknowledge message {messageId} failed because the channel was closed. The message will be requeued.");
-                    }
-                });
+            public IModel Channel { get; set; }
 
-                task.Start(exclusiveScheduler);
-                await task.ConfigureAwait(false);
-            }
-            catch (AlreadyClosedException ex)
-            {
-                Logger.Warn($"Attempt to acknowledge message {messageId} failed because the channel was closed. The message will be requeued.", ex);
-            }
+            public ulong DeliveryTag { get; set; }
         }
 
-        async Task RejectMessage(IModel channel, ulong deliveryTag, string messageId)
+        Task AcknowledgeMessage(IModel channel, ulong deliveryTag)
         {
-            try
+            return Task.Factory.StartNew(messageState =>
             {
-                var task = new Task(() =>
-                {
-                    if (channel.IsOpen)
-                    {
-                        channel.BasicReject(deliveryTag, true);
-                    }
-                    else
-                    {
-                        Logger.Warn($"Attempt to reject message {messageId} failed because the channel was closed. The message will be requeued.");
-                    }
-                });
+                var s = (MessageState)messageState;
 
-                task.Start(exclusiveScheduler);
-                await task.ConfigureAwait(false);
+                s.Channel.BasicAck(s.DeliveryTag, false);
 
-            }
-            catch (AlreadyClosedException ex)
+            }, new MessageState { Channel = channel, DeliveryTag = deliveryTag }, CancellationToken.None, TaskCreationOptions.DenyChildAttach, exclusiveScheduler);
+        }
+
+        Task RejectMessage(IModel channel, ulong deliveryTag)
+        {
+            return Task.Factory.StartNew(messageState =>
             {
-                Logger.Warn($"Attempt to reject message {messageId} failed because the channel was closed. The message will be requeued.", ex);
-            }
+                var s = (MessageState)messageState;
+
+                s.Channel.BasicReject(s.DeliveryTag, true);
+
+            }, new MessageState { Channel = channel, DeliveryTag = deliveryTag }, CancellationToken.None, TaskCreationOptions.DenyChildAttach, exclusiveScheduler);
         }
 
         public void Dispose()
```

Comment:
:+1:  It would be pretty easy to split this off into it's own PR, and I can make sure its split into a couple commits to make it clearer.